Closes #503: Primer for custom examples (#504)
pharmaverse/admiraldev: vignettes/constructing_custom_examples.Rmd

Reviewer: What about a couple of screenshots for the ToC, and a simple one of an example?  Just to give a bit of context in the guidance.

---
title: "Constructing Custom Examples"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 6
vignette: >
  %\VignetteIndexEntry{Constructing Custom Examples}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Introduction

> This vignette serves as a primer to help construct detailed, coherent and consistent 

Reviewer: ```suggestion
This vignette serves as a primer to help construct detailed and consistent 
```

Reviewer: just reducing the wordiage

Examples Sections in function documentation using the custom `rdx_roclet()` 
functionality - please see `demo_fun()` for a dummy example and  [`derive_extreme_records.R` in admiral](https://github.com/pharmaverse/admiral/blob/main/R/derive_extreme_records.R) 
for an example of this in action. 

> In practice, the rules and directives shown below cannot always be applied to the 

Reviewer: ```suggestion
In practice, the guidance shown below cannot always be applied to the 
```

letter, because each function and its context are different and may have varying 

> requirements. As such, developers are encouraged to treat these directives 
> heuristically and always apply their judgement when constructing Examples Sections.

Reviewer: ```suggestion
requirements. As such, developers are encouraged to apply their best judgement when constructing Examples Sections.
```

I don't know what heuristically means

Reviewer: im your audience - a peasant!

__why__ this may be a relevant scenario to consider). 
- The `@code` tag should be used to provide the code for the example. The code 
should be as self-contained as possible. The dataset/object being created should
be printed at the end of the section.

Note that a more complex block may also have multiple `@info` and`@code` 
sections after the `@caption` if it is particularly involved.

# How to sequence your blocks

Successive example blocks should build in complexity. The first examples should
showcase the most "minimal" execution of the function, with only the required
arguments being triggered and the default values discussed. The data in 
these examples is as concise as possible and may not be ADaM compliant, e.g.,
`STUDYID` may not be included or `AVISITN` may be present without `AVISIT`.

Later blocks should then bring in one/a few more arguments at a time as the
use-case becomes more involved. If possible, they should build on top of 
earlier blocks (i.e. use the same data but changing the scope of the function
call to add complexity), but if this is not possible, then new scenarios 
can be considered.

> All example blocks should answer questions like "what does this function do?"

Reviewer: oohhhh i like this - i feel like we should lead with this in the vignette
